flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/ExpressionParser.scala

```diff
@@ -45,6 +45,12 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
     ("""(?i)\Q""" + kw.key + """\E""").r
   }
 
+  // Convert the keyword into an case insensitive Parser
+  // It uses an more precisely matching mode for suffix keywords.
+  def suffixKeyword2Parser(kw: Keyword): Parser[String] = {
+    ("""(?i)\Q""" + kw.key + """\E(?![_$a-zA-Z0-9])""").r
```

Comment:
We can use this for all Keyword matches. Just replace `keyword2Parser` by this. 
This would be the only change in the `ExpressionParser`.

Reply:
We should match also Unicode letters, not just ASCII (`a-zA-Z`).